src/lib.rs

```diff
@@ -0,0 +1,436 @@
+                                    -> Result<JsonValue, KintoError> {
+
+        let endpoint = format!("{}/buckets/{bucket_id}/groups",
+                               self.server_url, bucket_id=bucket_id);
+        let payload = self.set_payload(data, permissions);
+        return self.handle_request(Method::Post, endpoint, payload.into());
+    }
+
+    pub fn update_group(&self, bucket_id: String, group_id: String, data: Option<JsonValue>)
+                         -> Result<JsonValue, KintoError> {
+        return self.update_group_with_perms(bucket_id, group_id, data, None);
+    }
+
+    pub fn update_group_with_perms(&self, bucket_id: String, group_id: String,
+                                    data: Option<JsonValue>,
+                                    permissions: Option<JsonValue>)
+                                    -> Result<JsonValue, KintoError> {
+
+        let endpoint = format!("{}/buckets/{bucket_id}/groups/{group_id}",
+                               self.server_url, bucket_id=bucket_id, group_id=group_id);
+        let payload = self.set_payload(data, permissions);
+        return self.handle_request(Method::Post, endpoint, payload.into());
+    }
+
+    pub fn patch_group(&self, bucket_id: String, group_id: String, data: Option<JsonValue>)
+                        -> Result<JsonValue, KintoError> {
+        return self.patch_group_with_perms(bucket_id, group_id, data, None);
+    }
+
+    pub fn patch_group_with_perms(&self, bucket_id: String, group_id: String,
+                                   data: Option<JsonValue>,
+                                   permissions: Option<JsonValue>)
+                                   -> Result<JsonValue, KintoError> {
+
+        let endpoint = format!("{}/buckets/{bucket_id}/groups/{group_id}",
+                               self.server_url, bucket_id=bucket_id, group_id=group_id);
+        let payload = self.set_payload(data, permissions);
+        return self.handle_request(Method::Post, endpoint, payload.into());
+    }
+
+
+    pub fn set_payload(&self, data: Option<JsonValue>,
+                       permissions: Option<JsonValue>) -> JsonValue {
+
+        let mut payload = JsonValue::new_object();
+
+        payload["data"] = data.unwrap_or(JsonValue::new_object());
+        payload["permissions"] = permissions.unwrap_or(JsonValue::new_object());
+
+        return payload;
+    }
+
+    pub fn handle_request(&self, method: Method, endpoint: String,
+                          payload: Option<JsonValue>)
+                          -> Result<JsonValue, KintoError> {
+
+        let request = self.http_client
+            .request(method, &endpoint)
+            .headers(self.headers.to_owned())
+            .body(payload.unwrap_or(JsonValue::new_object()).dump().as_str())
+            .send();
+
+        let mut response = match request {
+            Ok(response) => response,
+            Err(_) => return Err(KintoError::HyperError),
+        };
```

Comment:
This looks like a great place to use `try`. Also, why don't we expose the `HyperError` value rather than just the fact that it was a `HyperErorr`?